CHANGELOG.md

```diff
@@ -0,0 +1,5 @@
+# Changelog
+
+This changelog documents user-facing changes (features, fixes, and deprecations) in `modal`.
```

Comment:
> in _the Modal client library_.

How do you feel about the copy from https://github.com/modal-labs/modal-client/blob/387410ac6c5b860cde32f3cd8f96a9a16d77646d/CHANGELOG.md — users often ask us about our version support policy so it would be nice to include a sentence?

Reply:
Yeah good thinking! Where are you pulling the **Historical support matrix** from here? I think that @howard36 has been doing some work on defining support windows as well, I wonder how that intersects with this.

Reply:
Copied some of your header text over!

We might need to think a little more about the best way to programmatically sync the deprecation schedule that @howard36 implemented with the changelog, since it currently lives as a config in the `modal` repo. I've left that out for now. (I guess there's also a subtle distinction between "when will functionality be removed from the client" and "when will the server no longer support older clients").